netctx/packetconnctx.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-FileCopyrightText: 2023 The Pion community <https://pion.ly>
+// SPDX-License-Identifier: MIT
+
+package netctx
+
+import (
+	"context"
+	"io"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// ReaderFrom is an interface for context controlled packet reader.
+type ReaderFrom interface {
+	ReadFromContext(context.Context, []byte) (int, net.Addr, error)
+}
+
+// WriterTo is an interface for context controlled packet writer.
+type WriterTo interface {
+	WriteToContext(context.Context, []byte, net.Addr) (int, error)
+}
+
+// ReadFromWriterTo is a composite of ReadFromWriterTo.
+type ReadFromWriterTo interface {
+	ReaderFrom
+	WriterTo
+}
+
+// PacketConnCtx is a wrapper of net.PacketConn using context.Context.
+type PacketConnCtx interface {
```

Comment:
What do you think about removing the `Ctx` from the naming here? Isnt it implicitly given by the package name?

Reply:
I'm a fan! Only kept it for consistency with the existing `ConnCtx`, but since we are moving it anyway I think we can use `netctx.Conn` and `netctx.PacketConn` 👍🏻